agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -0,0 +1,77 @@
+package za.co.absa.atum.agent
+
+import org.apache.spark.sql.DataFrame
+import za.co.absa.atum.agent.model.{MeasureResult, Measurement}
+
+/**
+ *  AtumContext: This class provides the methods to measure Spark `Dataframe`. Also allows to add/edit/remove measures.
+ *  @param measurements: A sequences of measurements.
+ */
+
+case class AtumContext(measurements: Set[Measurement] = Set()) {
+
+  def withMeasures(
+    measure: Measurement
+  ): AtumContext =
+    this.copy(measurements = Set(measure))
```

Comment:
so this is more like 'withMeasureReplaced' ?

Reply:
yes, but also it creates the measures in case the measures don't exists